### PR TITLE
Framework update, package updates and async tweet

### PR DESCRIPTION
**Pull request by rickvdbosch**
- Migrated all projects to .NET Core 3.1 or .NET Standard 2.1
- Have frontend send message to queue to tweet instead of directly in the website
- Added Function to send Tweet based on queue message
- Migrated unit tests from NUnit to xUnit
- Updated NuGet packages
- Updated ARM templates based on recent changes

### CODE REVIEW

**Comment by jfversluis** [comment]
🤯 Thanks a bunch! Looks like the build failed though but that same build is nowhere to be found. Let me look into that.

**Comment by jfversluis** [comment]
OK, here it is: https://dev.azure.com/jfversluis/CfpExchange/_build/results?buildId=1134&view=logs&j=fd490c07-0b22-5182-fac9-6d67fe1e939b&t=ed931f23-ebb6-5c8f-76f6-22074363b910

> ##[error]The nuget command failed with exit code(1) and error(NU1201: Project CfpExchange.Models is not compatible with netcoreapp3.0 (.NETCoreApp,Version=v3.0). Project CfpExchange.Models supports: netstandard2.1 (.NETStandard,Version=v2.1)

I must be honest and note that I didn't pull down this code to try it locally yet. But seems like something is not updated yet in the Models?

**Comment by rickvdbosch** [comment]
When I looked at the details of the build right after it failed, I think I remember that the issue was the version of .NET Core. The pipeline is probably set up to use 2.2 explicitly. And I think failed builds are kept for 7 days by default, so that's probably the reason it's not there anymore 😁

**Comment by rickvdbosch** [comment]
Hey @jfversluis, I just updated one project to .NET Core 3.1 since it was targeting 3.0 instead 🤦🏻‍♂️. Close, but still no cigar.

I think the issue is the fact the build pipeline still uses `nuget.exe` instead of doing a `dotnet restore`. Can you change the build definition (or give me rights to do so 😜)?

**Comment by jfversluis** [comment]
Added you to the AzDO project so you should be able to do something yourself?

The dotnet restore works, but fails on something different now. I'm kind of tied up now, so to not block you... Go ahead!

**Comment by SebastianSchuetze** [comment]
Hey! Maybe you want to check this:

https://github.com/gulpjs/gulp/issues/2324

Gulp 3 and newer NodeJS do not work together. On the agent version 12 of nodejs is used probably.
Maybe you want to set the version of the tools you want to use as fixed tasks and install them before running anything.

**Comment by rickvdbosch** [comment]
Thanks @SebastianSchuetze, found that one too! Currently working on the build definition. Should be good to go somewhere today.

**Comment by rickvdbosch** [comment]
The build succeeded! Only thing to still keep into account:  
>Side note: I now reference a secret in Key Vault (as is the case with a similar setting) which isn't in there and doesn't look like it's added through ARM. So please take that into account 😄.
As found in [this comment](https://github.com/jfversluis/CfpExchange/issues/63#issuecomment-568443468) on issue #63 .

**Comment by SebastianSchuetze** [comment]
@rickvdbosch welcome. I suggested that I will convert the pipeline to YAML once you have merged yopur PR.

@jfversluis since YAML supports to have one pipeline for build / release. I would suggest that in the first phase I will create build only and then add the release pipeline.

File: CfpExchange/Middleware/RssMiddleware.cs [hunk 1]
```diff
@@ -54,14 +54,12 @@ public async Task Invoke(HttpContext context)
 				rssContent += "</channel>" +
 					"</rss>";
 
-				using (var memoryStream = new MemoryStream())
```
Comment (SebastianSchuetze) [inline CfpExchange/Middleware/RssMiddleware.cs]: Don't you want to export this to a method? You used that twice already. Or do you expect that these are the only two occurrences that will be in the solution?

**Comment by jfversluis** [comment]
Amazing job @rickvdbosch and @SebastianSchuetze thank you so much!

AFAIC this looks good to go for me.

@SebastianSchuetze 

> since YAML supports to have one pipeline for build / release. I would suggest that in the first phase I will create build only and then add the release pipeline.

Sounds lovely!

**Pull request merged**